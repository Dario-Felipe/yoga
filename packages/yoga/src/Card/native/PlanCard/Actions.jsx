import styled from 'styled-components';

import Actions from '../Card/Actions';
import theme from '../../../Theme/helpers/themeReader';

const Action = styled(Actions)`
  margin-top: ${theme.components.card.plan.actions.margin.top}px;
`;

Action.displayName = 'PlanCard.Actions';

export default Action;
