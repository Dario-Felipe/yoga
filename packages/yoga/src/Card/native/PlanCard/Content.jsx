import React from 'react';
import styled, { css } from 'styled-components';
import { string, shape, node } from 'prop-types';

import Content from '../Card/Content';
import Text from '../../../Text';
import theme from '../../../Theme/helpers/themeReader';

const Title = styled(Text.H5)`
  ${props => {
    const {
      components: {
        card: { plan },
      },
    } = theme(props);

    return css`
      color: ${plan.title.color};
      margin-bottom: ${plan.title.margin.bottom}px;
    `;
  }}
`;

const Description = styled(Text.Small)`
  height: 40px;

  color: ${theme.components.card.plan.description.color};
`;

const Price = styled.View`
  flex-direction: row;

  ${props => {
    const {
      components: {
        card: { plan },
      },
    } = theme(props);

    return css`
      margin-top: ${plan.price.margin.top}px;
      margin-bottom: ${plan.price.margin.bottom}px;
    `;
  }}
`;

const EnhancePrice = styled.View`
  ${({ align }) => `
    align-self: ${align};
  `}
`;

const Currency = styled(EnhancePrice)`
  margin-right: ${theme.components.card.plan.price.currency.margin.right}px;
`;

const PlanCardContent = ({
  title,
  description,
  currency,
  price,
  period,
  children,
  ...rest
}) => (
  <Content {...rest}>
    {title && <Title>{title}</Title>}
    {description && <Description numberOfLines={2}>{description}</Description>}
    <Price>
      {currency.prefix && (
        <Currency align="flex-start">
          <Text.Small>{currency.prefix}</Text.Small>
        </Currency>
      )}
      {price && <Text.H3>{price}</Text.H3>}
      {period && (
        <EnhancePrice align="flex-end">
          <Text.Small>{`${currency.suffix || ''}${period}`}</Text.Small>
        </EnhancePrice>
      )}
    </Price>
    {children}
  </Content>
);

PlanCardContent.propTypes = {
  /** Plan name */
  title: string.isRequired,
  /** currency of the current country */
  currency: shape({
    prefix: string,
    suffix: string,
  }).isRequired,
  price: string.isRequired,
  /** period that this price will be charged  */
  period: string.isRequired,
  description: string,
  children: node,
};

PlanCardContent.defaultProps = {
  children: null,
  description: null,
};

PlanCardContent.displayName = 'PlanCard.Content';

export default PlanCardContent;
