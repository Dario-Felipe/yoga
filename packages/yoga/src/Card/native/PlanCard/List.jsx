import React from 'react';
import styled, { css, withTheme } from 'styled-components';
import { View, TouchableWithoutFeedback } from 'react-native';
import { string, node, shape, oneOfType, func } from 'prop-types';

import Text from '../../../Text';
import theme from '../../../Theme/helpers/themeReader';

const List = styled.View`
  margin-top: ${theme.components.card.plan.list.margin.top}px;
`;

const Item = styled.View`
  margin-bottom: ${theme.components.card.plan.list.item.margin.bottom}px;
`;

const ItemText = styled(Text.Small)`
  ${props => {
    const {
      components: {
        card: { plan },
      },
    } = theme(props);

    return css`
      font-size: ${plan.list.item.font.size}px;
      color: ${plan.list.item.font.color};
    `;
  }}
`;

const Button = styled.View`
  margin-top: ${theme.components.card.plan.list.button.margin.top}px;
`;

const ButtonText = styled(Text.Medium)`
  ${props => {
    const {
      components: {
        card: { plan },
      },
    } = theme(props);

    return css`
      font-size: ${plan.list.button.font.size}px;
      color: ${plan.list.button.font.color};
    `;
  }}
`;

const ListItem = withTheme(
  ({
    text,
    icon: Icon,
    buttonProps: { children, ...buttonProps },
    theme: yogaTheme,
  }) => (
    <Item>
      <View style={{ flexDirection: 'row' }}>
        {Icon && (
          <Icon
            width={16}
            height={16}
            style={{
              marginRight:
                yogaTheme.yoga.components.card.plan.list.item.icon.margin.right,
            }}
          />
        )}
        <ItemText>{text}</ItemText>
      </View>
      {Boolean(Object.keys(buttonProps).length) && (
        <TouchableWithoutFeedback {...buttonProps}>
          <Button>
            <ButtonText>{children}</ButtonText>
          </Button>
        </TouchableWithoutFeedback>
      )}
    </Item>
  ),
);

List.displayName = 'PlanCard.List';
ListItem.displayName = 'PlanCard.ListItem';
Button.displayName = 'PlanCard.ListButton';

ListItem.propTypes = {
  text: string.isRequired,
  /** an icon to be displayed on the begin of the item */
  icon: oneOfType([node, func]),
  /** if provided displays a button below the item text. It accepts all button
   * element props */
  buttonProps: shape({}),
};

ListItem.defaultProps = {
  icon: undefined,
  buttonProps: {},
};

export { List, ListItem };
