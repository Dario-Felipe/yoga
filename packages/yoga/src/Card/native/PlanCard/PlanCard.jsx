import React from 'react';
import styled, { css } from 'styled-components';
import { node, string, oneOf } from 'prop-types';

import theme from '../../../Theme/helpers/themeReader';

import Card from '../Card';
import Text from '../../../Text';

const Plan = styled(Card)`
  max-width: 288px;
  width: 100%;

  position: relative;

  overflow: hidden;

  ${props => {
    const {
      colors,
      components: {
        card: { plan, elevation },
      },
    } = theme(props);

    return css`
      padding: ${plan.padding.top}px ${plan.padding.right}px
        ${plan.padding.bottom}px ${plan.padding.left}px;

      border-radius: ${plan.radius}px;

      background-color: ${colors.white};

      box-shadow: ${elevation};
    `;
  }}
`;

const Border = styled.View`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;

  height: 8px;
  background-color: ${({ variant }) =>
    theme.components.card.plan.colors[variant] ||
    theme.components.card.plan.colors.deep};
`;

const Ribbon = styled.View`
  position: absolute;
  align-items: center;
  justify-content: center;

  background-color: transparent;

  ${props => {
    const {
      components: {
        card: { plan },
      },
    } = theme(props);

    return css`
      top: ${plan.ribbon.position.top}px;
      left: ${plan.ribbon.position.left}px;

      padding-left: ${plan.ribbon.padding.left}px;
      padding-right: ${plan.ribbon.padding.right}px;

      border: ${plan.ribbon.border.width}px solid ${plan.ribbon.border.color};
      border-radius: ${plan.ribbon.radius}px;
    `;
  }}
`;
const RibbonText = styled(Text.Medium)`
  ${props => {
    const {
      components: {
        card: { plan },
      },
    } = theme(props);

    return css`
      font-size: ${plan.ribbon.font.size}px;
      line-height: ${plan.ribbon.font.height}px;
    `;
  }}
`;

const PlanCard = ({ children, ribbon, variant, ...rest }) => (
  <Plan {...rest}>
    <Border variant={variant} />
    {ribbon && (
      <Ribbon>
        <RibbonText>{ribbon}</RibbonText>
      </Ribbon>
    )}
    {children}
  </Plan>
);

PlanCard.propTypes = {
  children: node,
  /** A text to be displayed in a badge on top of card */
  ribbon: string,
  /** change the border top color */
  variant: oneOf([
    'vibin',
    'hope',
    'verve',
    'light',
    'energy',
    'medium',
    'stamina',
    'relax',
    'deep',
  ]),
};

PlanCard.defaultProps = {
  children: undefined,
  ribbon: undefined,
  variant: undefined,
};
PlanCard.displayName = 'PlanCard';

export default PlanCard;
