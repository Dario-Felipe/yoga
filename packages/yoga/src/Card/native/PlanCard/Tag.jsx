import React from 'react';
import styled, { css } from 'styled-components';
import { node } from 'prop-types';

import Text from '../../../Text';
import theme from '../../../Theme/helpers/themeReader';

const Wrapper = styled.View`
  position: absolute;
  align-items: center;
  justify-content: center;

  background-color: transparent;

  ${props => {
    const {
      components: {
        card: { plan },
      },
    } = theme(props);

    return css`
      top: ${plan.tag.position.top}px;
      left: ${plan.tag.position.left}px;

      padding-left: ${plan.tag.padding.left}px;
      padding-right: ${plan.tag.padding.right}px;

      border: ${plan.tag.border.width}px solid ${plan.tag.border.color};
      border-radius: ${plan.tag.radius}px;
    `;
  }}
`;
const StyledText = styled(Text.Medium)`
  ${props => {
    const {
      components: {
        card: { plan },
      },
    } = theme(props);

    return css`
      font-size: ${plan.tag.font.size}px;
      line-height: ${plan.tag.font.height}px;
    `;
  }}
`;

const Tag = ({ children, ...rest }) => (
  <Wrapper {...rest}>
    <StyledText>{children}</StyledText>
  </Wrapper>
);

Tag.propTypes = {
  children: node.isRequired,
};

export default Tag;
