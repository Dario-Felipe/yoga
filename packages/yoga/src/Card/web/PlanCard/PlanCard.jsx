import React from 'react';
import styled, { css } from 'styled-components';
import { node, oneOf, string } from 'prop-types';
import Box from '../../../Box';
import Text from '../../../Text';

export const PLAN_LINE_HEIGHT = 4;

const Plan = styled.article`
  width: 100%;
  display: flex;
  flex-direction: column;
  position: relative;
  max-width: 312px;

  overflow: hidden;

  ${props => {
    const theme = props.theme.yoga;
    const { plan } = theme.components.card;

    return css`
      padding: ${plan.padding.top}px ${plan.padding.right}px
        ${plan.padding.bottom}px ${plan.padding.left}px;
      border: ${theme.borders.small}px solid ${theme.colors.light};
      border-radius: ${plan.radius}px;

      background-color: ${theme.colors.white};
    `;
  }}
`;

const Border = styled.span`
  position: absolute;
  top: 0;
  left: 0;

  display: inline-block;
  width: 100%;

  height: ${PLAN_LINE_HEIGHT}px;
  background-color: ${({
    variant,
    theme: {
      yoga: {
        colors: { deepPurple, [variant]: color = deepPurple },
      },
    },
  }) => color};
`;

const DiscountWrapper = styled(Box).attrs({
  ph: 'xsmall',
  pv: 'xxxsmall',
})`
  ${({ bg, theme }) => {
    const MIN_WRAPPER_HEIGHT = 28;
    const baseCurveStyle = css`
      content: '';
      position: absolute;
      left: -${MIN_WRAPPER_HEIGHT}px;
      height: ${MIN_WRAPPER_HEIGHT}px;
      width: ${MIN_WRAPPER_HEIGHT}px;
    `;

    return css`
      position: absolute;
      top: 0;
      right: 0;
      display: flex;
      align-items: center;
      min-height: ${MIN_WRAPPER_HEIGHT}px;
      border-bottom-left-radius: 16px;

      ::before {
        ${baseCurveStyle}
        top: 0;
        background: ${theme.yoga.colors[bg]};
      }

      ::after {
        ${baseCurveStyle}
        top: ${PLAN_LINE_HEIGHT}px;
        background: ${theme.yoga.colors.white};
        border-top-right-radius: ${MIN_WRAPPER_HEIGHT / 2 - PLAN_LINE_HEIGHT}px;
      }
    `;
  }}
`;

function PlanCard({ children, discount, variant, ...rest }) {
  return (
    <Plan {...rest}>
      {discount && (
        <DiscountWrapper bg={variant}>
          <Text.Tiny color="white" fontWeight="medium">
            {discount}
          </Text.Tiny>
        </DiscountWrapper>
      )}
      <Border variant={variant} />
      {children}
    </Plan>
  );
}

PlanCard.propTypes = {
  children: node,
  discount: string,
  /** style the card border top color following the theme (primary, secondary,
   * vibin, hope, energy, relax, peace, verve, uplift, deepPurple, deep,
   * stamina, dark, medium, light, clear, white) */
  variant: oneOf([
    'primary',
    'secondary',
    'vibin',
    'hope',
    'energy',
    'relax',
    'peace',
    'verve',
    'uplift',
    'deepPurple',
    'stamina',
    'dark',
    'medium',
    'deep',
    'light',
    'clear',
    'white',
  ]),
};

PlanCard.defaultProps = {
  children: undefined,
  discount: undefined,
  variant: 'deepPurple',
};

PlanCard.displayName = 'PlanCard';

export default PlanCard;
