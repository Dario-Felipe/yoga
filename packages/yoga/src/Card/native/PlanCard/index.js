import PlanCard from './PlanCard';
import Content from './Content';
import Actions from './Actions';
import Subtitle from './Subtitle';
import { List, ListItem } from './List';

PlanCard.Content = Content;
PlanCard.Actions = Actions;
PlanCard.Subtitle = Subtitle;
PlanCard.List = List;
PlanCard.ListItem = ListItem;

export default PlanCard;
