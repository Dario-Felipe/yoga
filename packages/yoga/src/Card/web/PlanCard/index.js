import PlanCard from './PlanCard';
import Content from './Content';
import Actions from './Actions';
import Subtitle from './Subtitle';
import Tag from './Tag';
import { List, ListItem } from './List';

PlanCard.Content = Content;
PlanCard.Actions = Actions;
PlanCard.Subtitle = Subtitle;
PlanCard.Tag = Tag;
PlanCard.List = List;
PlanCard.ListItem = ListItem;

export default PlanCard;
