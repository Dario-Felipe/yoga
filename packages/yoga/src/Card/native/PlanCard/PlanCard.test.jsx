import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { Star } from '@gympass/yoga-icons';
import { ThemeProvider, Button } from '../../..';
import PlanCard from '.';

describe('<PlanCard />', () => {
  const buttonOnPressMock = jest.fn();

  const renderPlan = () =>
    render(
      <ThemeProvider>
        <PlanCard ribbon="Recommended Plan">
          <PlanCard.Content
            subtitle="plan"
            title="Basic"
            currency="$"
            price="99.90"
            period="/month"
          >
            <PlanCard.Subtitle>Get access to</PlanCard.Subtitle>
            <PlanCard.List>
              <PlanCard.ListItem
                icon={Star}
                text="list item"
                buttonProps={{
                  children: 'button',
                  onPress: buttonOnPressMock,
                }}
              />
            </PlanCard.List>
          </PlanCard.Content>
          <PlanCard.Actions>
            <Button full>Select this plan</Button>
          </PlanCard.Actions>
        </PlanCard>
      </ThemeProvider>,
    );
  describe('Events', () => {
    it('should call onPress when ListItem has a button', () => {
      const { getByText } = renderPlan();
      fireEvent.press(getByText('button'));

      expect(buttonOnPressMock).toHaveBeenCalled();
    });
  });

  describe('Snapshots', () => {
    it('should match snapshot with default PlanCard', () => {
      const { container: planCard } = renderPlan();
      expect(planCard).toMatchSnapshot();
    });
  });
});
