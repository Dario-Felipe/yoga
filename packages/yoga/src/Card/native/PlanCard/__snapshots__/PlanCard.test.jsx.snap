// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlanCard /> Snapshots should match snapshot with default PlanCard 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "elevation": 4,
          "shadowColor": "rgba(152, 152, 166, 0.25)",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 8,
            "elevation": 4,
            "overflow": "hidden",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 16,
            "maxWidth": 288,
            "overflow": "hidden",
            "paddingBottom": 24,
            "paddingLeft": 20,
            "paddingRight": 20,
            "paddingTop": 64,
            "position": "relative",
            "shadowColor": "rgba(152, 152, 166, 0.25)",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 6,
            "width": "100%",
          },
        ]
      }
      variant=""
      variantIntensity={3}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#d7d7e0",
              "height": 8,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#9898A6",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "justifyContent": "center",
              "left": 20,
              "paddingLeft": 8,
              "paddingRight": 8,
              "position": "absolute",
              "top": 20,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#41414A",
                "fontFamily": "Rubik",
                "fontSize": 12,
                "fontWeight": "500",
                "lineHeight": 16,
              },
            ]
          }
        >
          Recommended Plan
        </Text>
      </View>
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#6B6B78",
                "fontFamily": "Rubik",
                "fontSize": 12,
                "fontWeight": "500",
                "marginBottom": 8,
                "textTransform": "uppercase",
              },
            ]
          }
        >
          plan
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#001027",
                "fontFamily": "Rubik",
                "fontSize": 20,
                "fontWeight": "500",
                "lineHeight": 28,
                "marginBottom": 4,
              },
            ]
          }
        >
          Basic
        </Text>
        <Text
          numberOfLines={2}
          style={
            Array [
              Object {
                "color": "#6B6B78",
                "fontFamily": "Rubik",
                "fontSize": 14,
                "fontWeight": "400",
                "height": 40,
                "lineHeight": 20,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "marginBottom": 40,
                "marginTop": 32,
              },
            ]
          }
        >
          <View
            align="flex-start"
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#41414A",
                    "fontFamily": "Rubik",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "lineHeight": 20,
                  },
                ]
              }
            >
              $
            </Text>
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#41414A",
                  "fontFamily": "Rubik",
                  "fontSize": 48,
                  "fontWeight": "500",
                  "lineHeight": 56,
                },
              ]
            }
          >
            99.90
          </Text>
          <View
            align="flex-end"
            style={
              Array [
                Object {
                  "alignSelf": "flex-end",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#41414A",
                    "fontFamily": "Rubik",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "lineHeight": 20,
                  },
                ]
              }
            >
              /month
            </Text>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#6B6B78",
                "fontFamily": "Rubik",
                "fontSize": 12,
                "fontWeight": "500",
                "marginBottom": 8,
                "textTransform": "uppercase",
              },
            ]
          }
        >
          Get access to
        </Text>
        <View
          style={
            Array [
              Object {
                "marginTop": 16,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginBottom": 16,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <RNSVGSvgView
                bbHeight={16}
                bbWidth={16}
                fill="#001027"
                focusable={false}
                height={16}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "marginRight": 12,
                    },
                    Object {
                      "opacity": 1,
                    },
                    Object {
                      "flex": 0,
                      "height": 16,
                      "width": 16,
                    },
                  ]
                }
                width={16}
              >
                <RNSVGGroup
                  fill={
                    Array [
                      0,
                      4278194215,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  font={Object {}}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={0}
                  x={0}
                  y={0}
                />
              </RNSVGSvgView>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#001027",
                      "fontFamily": "Rubik",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "lineHeight": 20,
                    },
                  ]
                }
              >
                list item
              </Text>
            </View>
            <TouchableWithoutFeedback>
              <View
                style={
                  Array [
                    Object {
                      "marginTop": 12,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#652CDB",
                        "fontFamily": "Rubik",
                        "fontSize": 12,
                        "fontWeight": "500",
                      },
                    ]
                  }
                >
                  button
                </Text>
              </View>
            </TouchableWithoutFeedback>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "marginTop": 24,
            },
          ]
        }
      >
        <TouchableWithoutFeedback
          full={true}
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              disabled={false}
              full={true}
              inverted={false}
              pressed={false}
              small={false}
              style={
                Array [
                  Object {
                    "backgroundColor": "#F46152",
                    "borderRadius": 9999,
                    "height": 48,
                    "justifyContent": "center",
                    "paddingLeft": 24,
                    "paddingRight": 24,
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                disabled={false}
                inverted={false}
                pressed={false}
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontFamily": "Rubik",
                      "fontSize": 14,
                      "fontWeight": "700",
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Select this plan
              </Text>
            </View>
          </View>
        </TouchableWithoutFeedback>
      </View>
    </View>
  </View>
</View>
`;
